File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C++/s612878490_comments_removed.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int N, M;
    std::cin >> N >> M;

    std::vector<int> shortage;
    for (int i = 0; i < M; ++i) {
        int A;
        std::cin >> A;
        shortage.push_back(std::max(N - A, 0));
    }

    std::sort(shortage.begin(), shortage.end());

    int ans = 0;
    for (int i = 0; i < M - 1; ++i) {
        ans += shortage[i];
    }

    std::cout << ans << std::endl;
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Python/s612878490_comments_removed.py
import sys

line = input()
N, M = map(int, line.split())

shortage = []

for i in range(M):
    line = input()
    A = int(line.split()[0])
    shortage.append(max(N - A, 0))

shortage.sort()

ans = sum(shortage[:-1])

print(ans)
# End of Code
